pkg/util/cgroups/stats.go

```diff
@@ -0,0 +1,123 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+// +build linux
+
+package cgroups
+
+// PSIStats represent the Pressure Stall Information data
+// Source:
+// cgroupv1: not present
+// cgroupv2: *.pressure
+type PSIStats struct {
+	Avg10  *float64 // Percentage (0-100)
+	Avg60  *float64 // Percentage (0-100)
+	Avg300 *float64 // Percentage (0-100)
+	Total  *uint64  // Nanoseconds
+}
+
+// MemoryStats - all metrics in bytes except if otherwise specified
+// All statistics are hierarchical (i.e. includes all descendants)
+// Source:
+// cgroupv1: memory controller
+// cgroupv2: memory controller
+type MemoryStats struct {
+	UsageTotal   *uint64
+	Cache        *uint64
+	Swap         *uint64
+	RSS          *uint64
+	RSSHuge      *uint64
+	MappedFile   *uint64
+	Pgpgin       *uint64 // Number (no unit), cgroupv1 only
+	Pgpgout      *uint64 // Number (no unit), cgroupv1 only
+	Pgfault      *uint64 // Number (no unit)
+	Pgmajfault   *uint64 // Number (no unit)
+	InactiveAnon *uint64
+	ActiveAnon   *uint64
+	InactiveFile *uint64
+	ActiveFile   *uint64
+	Unevictable  *uint64
```

Comment:
Would be helpful to have few more details in comments here, aside from units.

Reply:
I'll add a link to refer to the Kernel doc describing the different fields.

Reply:
Thank you!